client/my-sites/upgrades/domain-management/email-forwarding/email-forwarding-add-new.jsx

```diff
@@ -49,30 +60,40 @@ const EmailForwardingAddNew = React.createClass( {
 		event.preventDefault();
 
 		if ( this.state.formSubmitting ) {
-			return false;
+			return;
 		}
 
 		this.setState( { formSubmitting: true } );
 
-		upgradesActions.addEmailForwarding( this.props.selectedDomainName, this.state.mailbox, this.state.destination, ( error ) => {
-			this.setState( { formSubmitting: false } );
-
-			this.recordEvent( 'addNewEmailForwardClick', this.props.selectedDomainName, this.state.mailbox, this.state.destination, ! Boolean( error ) );
-
-			if ( error ) {
-				notices.error( error.message );
-			} else {
-				this.props.successNotice( this.translate( 'Yay, %(email)s has been successfully added!', { args: {
-					email: this.state.mailbox + '@' + this.props.selectedDomainName
-				} } ) );
-
-				this.clearForm();
+		this.formStateController.handleSubmit( ( hasErrors ) => {
+			if ( hasErrors ) {
+				return;
 			}
+
+			const { mailbox, destination } = formState.getAllFieldValues( this.state.fields );
+
+			upgradesActions.addEmailForwarding( this.props.selectedDomainName, mailbox, destination, ( error ) => {
+				this.recordEvent( 'addNewEmailForwardClick', this.props.selectedDomainName, mailbox, destination, ! Boolean( error ) );
+
+				if ( error ) {
+					notices.error( error.message );
```

Comment:
You should check for the existence of `error.message`, because it might be empty/undefined (for example, when the request times out). So this line should be something like `notices.error( error.message || this.translate( 'Generic error message #2004' ) )`.